docs/building-modules/msg-services.md

```diff
@@ -0,0 +1,99 @@
+<!--
+order: 4
+-->
+# `Msg` Services
+
+A `Msg` Service processes [`message`s](./messages-and-queries.md#messages). `Msg` Services are specific to the module in which they are defined, and only process `message`s defined within the said module. They are called from `baseapp` during [`DeliverTx`](../core/baseapp.md#delivertx). {synopsis}
+
+## Pre-requisite Readings
+
+- [Module Manager](./module-manager.md) {prereq}
+- [Messages and Queries](./messages-and-queries.md) {prereq}
+
+## Implementation of a module `Msg` service
+
+All `Msg` processing is done by a `MsgServer`](messages-and-queries.md#msg-services) protobuf service. Each module should define a `Msg` service, which will be responsible for request and response serialization. 
+
+As further described in [ADR 031](../architecture/adr-031-msg-service.md), this approach has the advantages of clearly specifying return types and generating server and client code.
+
+When possible, the existing module's [`Keeper`](keeper.md) should implement `MsgServer`, else a `msgServer` struct that embeds the `Keeper` can be created, typically in `./keeper/msg_server.go`:
```

Comment:
```suggestion
When possible, the existing module's [`Keeper`](keeper.md) should implement `MsgServer`, otherwise a `msgServer` struct that embeds the `Keeper` can be created, typically in `./keeper/msg_server.go`:
```

not sure, "else" reads a bit weird to me